### PR TITLE
Switch to request

### LINKED ISSUE #6: isomorphic-fetch memory leak
It writes to global :(
https://github.com/matthew-andrews/isomorphic-fetch/blob/master/fetch-npm-node.js#L11-L16

https://github.com/matthew-andrews/isomorphic-fetch/issues/41

I'll make a PR to use request.

### PR DESCRIPTION
**Pull request by matteofigus**
Fixes #6

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -92,15 +92,15 @@ describe('OpenTable OC registry :: plugins :: graphql-plugin ', () => {
     it('should return a failure message', (done) => {
       client.query({ query: {}, variables: { test: 1 } }, { 'accept-language': 'en-US' })
         .catch(error => {
-          expect(error.message).to.equal('Failure')
```
Comment (matteofigus) [inline tests/index.js:95]: There is no status code that translates to "Failure"

File: index.js [hunk 2]
```diff
@@ -1,51 +1,37 @@
...
 module.exports.execute = () => ({
-  query: (options, headers, timeout) => client(options, headers, timeout),
+  query: (options, headers, timeout) => new Promise((resolve, reject) => request({
+    body: {
+      query: options.query,
+      variables: options.variables,
+      operationName: null,
+    },
+    headers: _.extend({ 'User-Agent': 'oc' }, headers),
+    json: true,
```
Comment (matteofigus) [inline index.js:24]: json true makes unnecessary to send json header, stringify body, and parse response

File: tests/index.js [hunk 3]
```diff
@@ -92,15 +92,15 @@ describe('OpenTable OC registry :: plugins :: graphql-plugin ', () => {
...
+          expect(error.message).to.equal('Internal Server Error')
         }).then(done, done)
     });
   });
 
   describe('when calling query successfully', () => {
     let client;
     const plugin = injectr('../index.js', {
-      'isomorphic-fetch': sinon.stub().returns(Promise.resolve({ ok: true, json: () => Promise.resolve('PASSED') }))
```
Comment (matteofigus) [inline tests/index.js:103]: we assume the response is json, not a string

File: index.js [hunk 4]
```diff
@@ -1,51 +1,37 @@
...
-
-const parseResponse = response => response.json();
-
-module.exports.register = (opts, dependencies, next) => { // eslint-disable-line consistent-return
+module.exports.register = (opts, dependencies, next) => {
   if (!opts.serverUrl) {
     return next(new Error('The serverUrl parameter is invalid'));
   }
 
-  client = (options, headers, timeout) => fetch(opts.serverUrl, {
```
Comment (matteofigus) [inline index.js:33]: moving this to the execute() as that gets garbage collected on any component's domain execution is finished

**Review by chriscartlidge — APPROVED** [review]

**Comment by chriscartlidge** [comment]
Good catch @matteofigus 👍

**Pull request merged**